lib/RTCSession.js

```diff
@@ -1046,13 +1047,32 @@ module.exports = class RTCSession extends EventEmitter
       }
       else
       {
-        const dtmf = new RTCSession_DTMF(this);
+        // send DTMF according to transport config
+        switch (transportType)
+        {
+          case JsSIP_C.DTMF_TRANSPORT.IN_BAND: {
+            // send DTMF in current audio RTP stream
+            const sender = this._getDTMFRTPSender();
+
+            if (sender)
+            {
+              sender.insertDTMF(tone, duration, interToneGap);
+            }
 
-        options.eventHandlers = {
-          onFailed : () => { this._tones = null; }
-        };
-        dtmf.send(tone, options);
-        timeout = duration + interToneGap;
+            break;
+          }
+          case JsSIP_C.DTMF_TRANSPORT.INFO:
+          default: {
```

Comment:
Don't use `default`. `transportType` must be validated above.

```suggestion
          case JsSIP_C.DTMF_TRANSPORT.INFO:
          {
```